contracts/proxy-deployer/src/contract_interactions.rs

```diff
@@ -95,6 +107,66 @@ pub trait ContractInteractionsModule: config::ConfigModule + pause::PauseModule
             .execute_on_dest_context()
     }
 
+    /// Use this endpoint to transfer the ownership
+    /// This is needed to properly update the stored data
+    #[endpoint(changeOwnerAddress)]
+    fn change_owner(
+        &self,
+        contract_address: ManagedAddress,
+        new_owner: ManagedAddress,
+        opt_orig_owner: OptionalValue<ManagedAddress>,
+    ) {
+        self.can_call_endpoint(Some(contract_address.clone()));
+        require!(
+            self.blockchain().is_smart_contract(&contract_address),
+            "Contract address is not a SC"
+        );
+        require!(
+            !self.blacklisted_deployers_list().contains(&new_owner),
+            "New owner is blacklisted"
+        );
+        let mut caller = self.blockchain().get_caller();
+        if caller == self.blockchain().get_owner_address() {
+            require!(opt_orig_owner.is_some(), "Must provide original owner");
+            caller = opt_orig_owner.into_option().unwrap();
+        }
+        let contract_template_mapper = self.contract_template(&contract_address);
+        require!(!contract_template_mapper.is_empty(), "No template found");
+
+        let mut contract_processed = false;
+        let template_address = contract_template_mapper.take();
+        let deployer_template_addresses_mapper = self.deployer_template_addresses(&caller);
+        let mut deployer_template_addresses = deployer_template_addresses_mapper
+            .get(&template_address)
+            .unwrap();
+        for index in 0..deployer_template_addresses.len() {
+            let deployed_address = deployer_template_addresses.get(index).clone_value();
+            if deployed_address == contract_address {
+                deployer_template_addresses.remove(index);
+                self.deployer_template_addresses(&caller)
+                    .insert(template_address, deployer_template_addresses);
+
+                contract_processed = true;
+                break;
+            }
+        }
+
+        require!(contract_processed, "Contract not found for deployer");
+        self.deployer_contracts(&caller)
+            .swap_remove(&contract_address);
+        if self.deployer_contracts(&caller).is_empty() {
+            self.deployers_list().swap_remove(&caller);
+        }
+
+        let () = self
```

Comment:
I don't think you need to specify it has no return value?

Reply:
It doesn't let me to compile without it, as it throws the error: `cannot satisfy `_: TopDecodeMulti``